packages/allow-scripts/src/index.js

```diff
@@ -1,671 +1,30 @@
- */
-function prepareBinScriptsPolicy(binCandidates) {
-  /** @type {Record<string, string>} */
-  const policy = {}
-  // pick direct dependencies without conflicts and enable them by default unless colliding with bannedBins
-  for (const [bin, infos] of binCandidates.entries()) {
-    const binsFromDirectDependencies = infos.filter((i) => i.isDirect)
-    if (binsFromDirectDependencies.length === 1 && !bannedBins.has(bin)) {
-      // there's no conflicts, seems fairly obvious choice to put it up
-      policy[bin] = binsFromDirectDependencies[0].fullLinkPath
-    }
-  }
-  return policy
-}
-
-/**
- * @param {BinsConfig} param0
- */
-function printPackagesByBins({ allowedBins, excessPolicies }) {
-  console.log('\n# allowed packages with bin scripts')
-  if (allowedBins.length) {
-    allowedBins.forEach(({ canonicalName, bin }) => {
-      console.log(`- ${canonicalName} [${bin}]`)
-    })
-  } else {
-    console.log('  (none)')
-  }
-
-  if (excessPolicies.length) {
-    console.log(
-      '\n# packages with bin scripts that no longer need configuration (package or script removed or script path outdated)'
-    )
-    excessPolicies.forEach((bin) => {
-      console.log(`- ${bin}`)
-    })
-  }
-}
-
-/**
- * @param {ScriptsConfig} param0
- */
-function printPackagesByScriptConfiguration({
-  packagesWithScripts,
-  allowedPatterns,
-  disallowedPatterns,
-  missingPolicies,
-  excessPolicies,
-}) {
-  console.log('\n# allowed packages with lifecycle scripts')
-  if (allowedPatterns.length) {
-    allowedPatterns.forEach((pattern) => {
-      const collection = packagesWithScripts.get(pattern) || []
-      console.log(`- ${pattern} [${collection.length} location(s)]`)
-    })
-  } else {
-    console.log('  (none)')
-  }
-
-  console.log('\n# disallowed packages with lifecycle scripts')
-  if (disallowedPatterns.length) {
-    disallowedPatterns.forEach((pattern) => {
-      const collection = packagesWithScripts.get(pattern) || []
-      console.log(`- ${pattern} [${collection.length} location(s)]`)
-    })
-  } else {
-    console.log('  (none)')
-  }
-
-  if (missingPolicies.length) {
-    console.log('\n# unconfigured packages with lifecycle scripts')
-    missingPolicies.forEach((pattern) => {
-      const collection = packagesWithScripts.get(pattern) || []
-      console.log(`- ${pattern} [${collection.length} location(s)]`)
-    })
-  }
-
-  if (excessPolicies.length) {
-    console.log(
-      '\n# packages with lifecycle scripts that no longer need configuration due to package or scripts removal'
-    )
-    excessPolicies.forEach((pattern) => {
-      const collection = packagesWithScripts.get(pattern) || []
-      console.log(`- ${pattern} [${collection.length} location(s)]`)
-    })
-  }
-}
-
-/**
- * @param {SavePackageConfigurationsOpts} param0
- * @returns {Promise<void>}
- */
-async function savePackageConfigurations({
-  rootDir,
-  conf: {
-    packageJson,
-    configs: { lifecycle, bin },
+  setup: {
```

Comment:
Note: This keeps the `setup` export identical to before, instead of exporting some functions twice. If we want to also change the external API, we can do so in a follow-up.